pkg/descheduler/evictions/evictions.go

```diff
@@ -110,9 +110,9 @@ func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node,
 	if len(reasons) > 0 {
 		reason += " (" + strings.Join(reasons, ", ") + ")"
 	}
-	if pe.maxPodsToEvictPerNode > 0 && pe.nodepodCount[node]+1 > pe.maxPodsToEvictPerNode {
```

Comment:
I believe this was checking `>0` because the default of `0 = unlimited`. So, removing this and making nil the default changes the existing behavior. I think we should still respect 0 as no limit

Reply:
Based on https://github.com/kubernetes-sigs/descheduler#policy-and-strategies:

The policy includes a common configuration that applies to all the strategies:
| Name | Default Value | Description |
|------|---------------|-------------|
| `nodeSelector` | `nil` | limiting the nodes which are processed |
| `evictLocalStoragePods` | `false` | allows eviction of pods with local storage |
| `evictSystemCriticalPods` | `false` | [Warning: Will evict Kubernetes system pods] allows eviction of pods with any priority, including system pods like kube-dns |
| `ignorePvcPods` | `false` | set whether PVC pods should be evicted or ignored |
| `maxNoOfPodsToEvictPerNode` | `nil` | maximum number of pods evicted from each node (summed through all strategies) |

`maxNoOfPodsToEvictPerNode` is nil by default and there's no mention of 0 being the unlimited case. Also, using 0 for unlimited is not intuitive since without any other information setting it to 0 is like saying "do not evict any pod from any node".

Reply:
Ah okay, I was looking at the componentconfig types where it wasn't a pointer. So, 0 was just unlimited for the internal representation. Sounds good